src/components/pages/info-pages/ClassInfoPage.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react';
+import { getClassDetails, getUserData } from '../../../util/DataBaseRequests';
+import { useAuth } from '../../../AuthProvider';
+import { useParams } from 'react-router-dom';
+import Loader from '../../common/Loader';
+import AgeIcon from '../../../assets/icons/icon-age.svg';
+import LessonTypeIcon from '../../../assets/icons/icon-lesson.svg';
+import ScheduleIcon from '../../../assets/icons/icons-schedule.svg';
+import IconClock from '../../../assets/icons/icon-clock.svg';
+import IconTypeLesson from '../../../assets/icons/icon-type.png';
+import TeacherInfo from './TeacherInfo';
+
+const ClassInfoPage = () => {
+  const { userData } = useAuth();
+  const { classId } = useParams();
+  const [classItem, setClassItem] = useState(null);
+  const [teacherInfo, setTeacherInfo] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    if (userData) {
+      const fetchClassAndTeacher = async () => {
+        try {
+          // Fetch the class details
+          const classData = await getClassDetails(classId, userData.token);
+          setClassItem(classData.class);
+
+          // Fetch the teacher details
+          if (classData.class.createdBy) {
+            const teacherData = await getUserData(
+              classData.class.createdBy,
+              userData.token
+            );
+            setTeacherInfo(teacherData.data);
+          }
+        } catch (error) {
+          setError(error.message);
+        } finally {
+          setIsLoading(false);
+        }
+      };
+
+      fetchClassAndTeacher();
+    } else {
+      error('User should log in.');
+    }
+  }, [classId, userData]);
+
+  if (isLoading) return <Loader />;
+  if (error) return <p>Error: {error}</p>;
+  if (!classItem) return <p>Class not found.</p>;
+
+  return (
+    <div className="container m-auto w-10/12">
+      <div className="flex flex-col w-full text-black">
+        <div className="flex justify-between flex-col lg:flex-row">
+          <div className="flex flex-col w-full lg:w-[75%] md:flex-row lg:flex-row pb-4">
+            <div className="w-full m:w-[60%] font-roboto text-xl">
+              <h2 className="font-spartan font-medium text-4xl pb-8">
+                {classItem.classTitle}
+              </h2>
+              <p className="w-full md:w-[95%]">{classItem.description}</p>
+            </div>
+
+            <div className="flex w-full lg:w-3/4 pl-0 md:pl-2">
```

Comment:
Please, fix picture size
May just leave w-full

W/o this changes now it looks so on my side:
<img width="1467" alt="Screenshot 2024-09-02 at 6 03 13 PM" src="https://github.com/user-attachments/assets/76cbf8dd-20fc-4698-88ef-d6f4bed49908">